experiment/build/local_build.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module for building things on Google Cloud Build for use in trials."""
+
+import os
+import posixpath
+from typing import Tuple
+
+from common import benchmark_utils
+from common import environment
+from common import experiment_path as exp_path
+from common import gsutil
+from common import logs
+from common import new_process
+from common import utils
+from experiment.build import build_utils
+
+# Maximum time to wait for a GCB config to finish build.
+GCB_BUILD_TIMEOUT = 2 * 60 * 60  # 2 hours.
+
+# High cpu configuration for faster builds.
+GCB_MACHINE_TYPE = 'n1-highcpu-8'
+
+logger = logs.Logger('builder')  # pylint: disable=invalid-name
+
+
+def make(*args):
+    """Invoke |make| with |args| and return the result."""
+    assert args
+    command = ['make'] + list(args)
+    return new_process.execute(command, cwd=utils.ROOT_DIR)
+
+
+def build_base_images() -> Tuple[int, str]:
+    """Build base images on GCB."""
+    return make('base-runner', 'base-builder')
+
+
+def build_coverage(benchmark):
+    """Build coverage image for benchmark on GCB."""
+    image_name = 'build-coverage-{}'.format(benchmark)
+    result = make(image_name)
+    if result.retcode:
+        return result
+    local_copy_coverage_binaries(benchmark)
+    return result
+
+
+def local_copy_coverage_binaries(benchmark):
+    """Copy coverage binaries in a local experiment."""
+    shared_volume = os.environ['SHARED_VOLUME']
```

Comment:
What is this, needs to come via some experiment config attribute, not a env variable (undocumented).

Reply:
The problem with using experiment configs is the way we currently do it, they have to be plumbed deep into the depths of the code to be useful. I may switch this to a model where we have a config module that caches the config and can be queried from anywhere but right now, I think env vars are an acceptable alternaitve.